app/policies/events_policy.rb

```diff
@@ -3,21 +3,17 @@
 class EventsPolicy < CollectionPolicy
 
   def index?
-    user.known?
+    PackagesPolicy.new(user,scope).index?
   end
 
   def base_scope
     Event.all
   end
 
   def resolve
-    if user.admin?
-      scope.all
-    elsif user.known?
-      scope.owned(user.id)
-    else
-      scope.none
-    end
+    # n.b. the base scope is Event.all here, but since Event has the scame
+    # scopes as Packages this will still work as intended.
+    PackagesPolicy.new(user,scope).resolve
```

Comment:
This is pretty sketchy. I think what we're trying to express here is that you can view Events for any package you can view. Related to the comment above, I think we're probably looking for

```
scope.for_packages(PackagesPolicy.new(user).resolve)
```

Where `Event.for_packages` is:

```
scope :for_packages, ->(package_scope) { joins(:package).merge(package_scope) }
```

Reply:
There is a problem with this suggestion in that we inject a scope of FakeCollection.all. In this usage, `PackagesPolicy.new(user)` gets its default scope of `Package.all`, and so ActiveRecord and relations junk starts pouring into the policy specs.

I am increasingly thinking it makes sense to do away with the event & queue items policy altogether, have the controllers use the package policy, and leave it to the models to do the right thing. If this ends up being painful later, we can certainly change it, but right now `EventsPolicy` and `QueueItemsPolicy` really do basically nothing except enforce an artificial 1:1 relationship between models, controllers, and policies.

Reply:
Ahh, I see... I don't see how we would do the delegation without encountering AR short of a named parameter for the package scope (or the PackagesPolicy)... Kinda gross. I suppose it's a collaborator either way, so it makes sense in the big picture, but I'm not sure we want/need to do it.

Reply:
See the latest commit, which just uses `PackagesPolicy` directly.